src/cmd/services/m3aggregator/serve/serve.go

```diff
@@ -54,6 +62,21 @@ func Serve(
 	defer httpServer.Close()
 	log.Infof("http server: listening on %s", httpAddr)
 
+	if m3msgServerOpts != nil {
```

Comment:
The code **never** skips this `if` (the way its structured right now)

The options passed to this are created by the following line in `src/cmd/services/m3aggregator/main/main.go `:

```
m3msgAddr, m3msgServerOpts := m3msg.NewServerOptions(iOpts, cfg.M3msg)
```

Which is the following method (from `src/aggregator/server/m3msg/options.go`): 

```
// NewServerOptions creates a new set of m3msg server options.
func NewServerOptions(
	iOpts instrument.Options,
	m3msgConfiguration *coordinatorM3msg.Configuration,
) (string, Options) {
	var addr string
	if m3msgConfiguration != nil {
		addr = m3msgConfiguration.Server.ListenAddress
	}
	return addr, &options{
		instrumentOpts:     iOpts,
		m3msgConfiguration: m3msgConfiguration,
	}
}
```

i.e. options are never nil. Would be better to add an `Enabled()` on the options to indicate whether the server is actually enabled.

Reply:
Hm actually, is the m3msg server supposed to be optional?